hooks/pre_gen_project.py

```diff
@@ -0,0 +1,12 @@
+"""
+    This script has as purpose to create the Python package name and the 
+    'Inmanta module name' using the module name inputed by the user (or the default module name)
+    The Python package name uses the same naming convention as standart Python packages: 
+    all lowercase and words separated by "_".
+    The 'Inmanta module name' (which is also the name of the Python module name) uses "-" to separate
+    words and can have uppercase letters except for the first letter which is always lowercase.
+
+"""
+
+{{ cookiecutter.update({"package_name": "inmanta-module-" + cookiecutter.module_name.lower().replace("_","-") }) }}
+{{ cookiecutter.update({"inmanta_module_name": (cookiecutter.module_name[0].lower() + cookiecutter.module_name[1:]).replace("-","_") }) }}
```

Comment:
This will fail to convert the first character if it happens to be a `-`. But if the user inputs something that exotic they deserve what's coming imo.

Reply:
maybe like this?

```
{{ cookiecutter.update({"inmanta_module_name": (cookiecutter.lstrip(string.punctuation).module_name[0].lower() + cookiecutter.lstrip(string.punctuation).module_name[1:]).replace("-","_") }) }}
```

I did not find how to use variables declared out off the scope of the {{ }} in the {{ }}, which would help make it more readable :/

Reply:
Actually, I wouldn't pay too much attention to this, I'm pretty sure a module named `_my_module` would be rejected by the compiler as well so I think for now we can just keep assuming the user will input something at least somewhat sane. You can leave this as you initially wrote it in my opinion.